netbox/project-static/js/forms.js

```diff
@@ -221,7 +221,7 @@ $(document).ready(function() {
                 var results = data.results;
 
                 results = results.reduce((results,record,idx) => {
-                    record.text = record[element.getAttribute('display-field')] || record.name;
+                    record.text = record[element.getAttribute('display-field')] || record.name || record.model;
```

Comment:
Instead of appending individual fields, we should ensure that the `display-field` attribute is being set on the device type form field.